dev/interface_definitions.h

```diff
@@ -0,0 +1,359 @@
+/** @file Mainly existing to disentangle implementation details from circular and cross dependencies
+ *  (e.g. column_t -> default_value_extractor -> serializator_context -> storage_impl -> table_t -> column_t)
+ *  this file is also used to provide definitions of interface methods 'hitting the database'.
+ */
+#pragma once
+#include <sqlite3.h>
+#include <type_traits>  //  std::is_same
+#include <utility>  //  std::move
+#include <algorithm>  //  std::find_if
+#include <sstream>  //  std::stringstream
+#include <cstdlib>  //  std::atoi
+
+#include "error_code.h"
+#include "default_value_extractor.h"
+#include "type_printer.h"
+#include "column.h"
+#include "table.h"
+#include "storage_impl.h"
+#include "storage.h"
+#include "dbstat.h"
+#include "util.h"
+
+namespace sqlite_orm {
+    namespace internal {
+
+        template<class O, class T, class G, class S, class... Op>
+        std::unique_ptr<std::string> column_t<O, T, G, S, Op...>::default_value() const {

```

Comment:
> /** Mainly existing to disentangle implementation details from circular and cross dependencies 
>     (e.g. column_t -> default_value_extractor -> serializator_context -> storage_impl -> table_t -> column_t) 
>     this file is also used to provide definitions of interface methods 'hitting the database'. 
>  */

Reply:
ok I see. Ok let's leave it like this. Thanks. I'll run this branch on my mac and if everything is ok then I'll merge it

Reply:
<img width="1391" alt="Снимок экрана 2022-03-25 в 00 15 30" src="https://user-images.githubusercontent.com/6009640/159983668-2e9dfe5f-9ba3-4c0d-8ed9-77d9f30221b8.png">
looks like the problem in position of defining `storage_impl<>` specialization:

- problem line is 2011
- specialization is defined on line 10k+

<img width="604" alt="Снимок экрана 2022-03-25 в 00 16 26" src="https://user-images.githubusercontent.com/6009640/159983823-edcd92c6-0ed0-4e16-93e4-b9f226247ee5.png"> 

@trueqbit can you repro it?

Reply:
We could split the definitions into separate implementation header files, and the include them all in the one _interface_definitions.h_ header file. This way it would be at least cleanly separated.

Reply:
ok please go ahead. I fully trust you

Reply:
I'll do it, then I also split the implementations at the same time.